source/features/repo-age.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import * as icons from '../libs/icons';
+import {getRepoGQL, timeAgo} from '../libs/utils';
+
+async function getRepoCreationDate(): Promise<Date> {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			createdAt
+		}
+	`);
+
+	return new Date(repository.createdAt);
+}
+
+async function init(): Promise<void> {
+	const date = await getRepoCreationDate();
+	const {interval, unit} = timeAgo(date);
+	const element = (
+		<li title={`Repository created ${date.toDateString()}`}>
+			{icons.calendar()}
+
+			<span className="num text-emphasized">{interval}</span> {unit} old
```

Comment:
GitHub has their own relative time element and we should use that one instead of having our own “time ago” function

Reply:
I didn't realize that, sweet. I'm not super familiar with web components - I added the ```time-ago``` time element to the IntrinsicElements in _globals.d.ts_ interface and have it working, but how would you recommend I separate the value into an emphasized <span> so it's 

**1** year ago or **2** hours ago

Should I extend the component?

Reply:
Alright, maybe not. But you can replace the current code with `Intl`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RelativeTimeFormat

It would probably become:

```Js
const [interval, unit] = rtf.format(date).split(' ');
```

Reply:
As far as I can tell, ```Intl.RelativeTimeFormat()``` is only for language localization of the relative time units. You can't pass it a date to compute difference, instead you pass it an already decided interval and unit of time so I'll still need some kind of interval difference logic.

Reply:
D'oh, indeed. I just wonder if there's a way to shorten that code (`timeAgo`)

I'll look into it